### PR TITLE
[feat] Join Class Functionality

### PR DESCRIPTION
**Pull request by jjstnlee**
[//]: # "Feel free to customize this template and the emojis to your project's vibes"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

- implemented join class functionality :3
- fixed styling for join class page

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 
Screenshot 2025-11-02 at 16 46 58
Screenshot 2025-11-02 at 16 47 14

## 👍 How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

## 👍 Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @nehaahussain

### CODE REVIEW

File: app/join/page.tsx [hunk 1]
```diff
@@ -1,21 +1,76 @@
...
 
 export default function JoinPage() {
   const router = useRouter();
+  const [joinCode, setJoinCode] = useState("");
+  const [groups, setGroups] = useState<Group[]>([]);
+  const [error, setError] = useState("");
+
+  useEffect(() => {
+    async function fetchGroups() {
+      const localData = await fetchLocalDatabase();
```
Comment (nehaahussain) [inline app/join/page.tsx]: we are calling `fetchLocalDatabase()` twice — once in `app/join/page.tsx` and another in `app/lessons/[groupId]/page.tsx`. could this be redundant/inefficient since we're reloading the same local data twice when it could be shared instead?

perhaps we could fetch the local database once then store it in context. this may be a better approach since we already have multiple pages reading from the same local DB and will probably add more. what do you think?

if we go this route, the react context would include these steps:
- fetches fetchLocalDatabase() once (in a provider component)
- stores local data
- lets both join page and lessons page access it.

File: app/lessons/[groupId]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      // Fetch lessons for the given groupId from local database
+      const localData = await fetchLocalDatabase();
+      const lessons = localData.lessons.filter(
+        lesson => lesson.group_id.toString() === groupId,
+      );
+      setLessons(lessons);
+    }
+    fetchLessons();
+  }, []);
+
```
Comment (nehaahussain) [inline app/lessons/[groupId]/page.tsx]: let's make sure to add loading or empty states incase there's no data yet or at all.

`if (!lessons.length) return <p>No lessons found.</p>;`

File: app/join/page.tsx [hunk 3]
```diff
@@ -1,21 +1,76 @@
...
+    }
+    fetchGroups();
+  }, []);
+
+  const joinCodes = groups.map(group => group.join_code);
+
+  function handleJoin(e: React.MouseEvent<HTMLButtonElement, MouseEvent>) {
+    e.preventDefault();
+
+    if (!joinCode || !joinCodes.includes(joinCode)) {
```
[the remark this answers is not in the document]
Comment (nehaahussain) [reply app/join/page.tsx]: agreed. let's make the functionality case insensitive so that inputting "FARMING" and "farming" will both work.

**Review by nehaahussain — CHANGES_REQUESTED** [review]
GREAT JOB JULEE! added some comments so plz slack me when addressed but looks so good.
